### PR TITLE
Update JSON package collection models

### PR DESCRIPTION
**Pull request by yim-lee**
According to https://forums.swift.org/t/package-collection-format/42071

- Rename `summary` and `title` to `description` and `name`, respectively
- Add `keywords` to `PackageCollection.Package`
- Add `minimumPlatformVersions` to `PackageCollection.Package.Version`
- Add `createdBy` to `PackageCollection`

API models and tests are updated as well.

### CODE REVIEW

File: Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift [hunk 1]
```diff
@@ -16,6 +16,8 @@ import struct Foundation.URL
 import PackageModel
 import SourceControl
 
+fileprivate typealias JSONModel = JSONPackageCollectionModel.V1
```
Comment (tomerd) [inline Sources/PackageCollections/Providers/JSONPackageCollectionProvider.swift:19]: nice

**Comment by tomerd** [comment]
looks good from a quick glance. lets get  #3068 and  #3069 merged first as there is tiny overlap

note: need to add new files to CMakeLists in PackageCollections directory

File: Sources/PackageCollections/JSONModel/JSONCollection+v1.swift [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+extension JSONPackageCollectionModel.V1 {
+    public struct PackageCollection: Equatable, Codable {
+        /// The name of the package collection, for display purposes only.
+        public let name: String
+
+        /// A description of the package collection.
+        public let _description: String?
+
+        /// An array of keywords that the collection is associated with.
+        public let keywords: [String]?
```
Comment (abertelrud) [inline Sources/PackageCollections/JSONModel/JSONCollection+v1.swift:29]: Is this optional so it doesn't get coded at all if empty, or is there a semantic difference?  (noting for example that `packages` below isn't optional — though admittedly we don't expect to have no packages the way there might be no keywords)

**Review by abertelrud — APPROVED** [review]

**Comment by tomerd** [comment]
@yim-lee please rebase

File: Sources/PackageCollections/JSONModel/JSONCollection+v1.swift (hunk 2 above)
Comment (yim-lee) [reply Sources/PackageCollections/JSONModel/JSONCollection+v1.swift:29]: It is optional so it doesn't get coded if `nil` and deserialization doesn't fail if the key is missing. `[]` still gets coded.

`packages` is required. Decode will fail if the key is missing or value is `nil`. `[]` will get through and we will probably need validation to trap that case.

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageCollections/JSONModel/JSONCollection+v1.swift (hunk 2 above)
Comment (abertelrud) [reply Sources/PackageCollections/JSONModel/JSONCollection+v1.swift:29]: The distinction between optional and required arrays makes a lot of sense.  Thanks for explaining!

**Review by abertelrud — APPROVED** [review]

File: Sources/PackageCollections/JSONModel/JSONCollection+v1.swift [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+import struct Foundation.URL
+
+import PackageModel
+
+extension JSONPackageCollectionModel {
+    public enum V1 {}
+}
+
+extension JSONPackageCollectionModel.V1 {
+    public struct PackageCollection: Equatable, Codable {
```
Comment (tomerd) [inline Sources/PackageCollections/JSONModel/JSONCollection+v1.swift:21]: consider PackageCollection -> Collection

File: Sources/PackageCollections/Model/Collection.swift [hunk 4]
```diff
@@ -113,6 +119,14 @@ extension PackageCollectionsModel {
     }
 }
 
+extension PackageCollectionsModel {
+    /// Represents the author of a `PackageCollection`
+    public struct CollectionAuthor: Equatable, Codable {
```
Comment (tomerd) [inline Sources/PackageCollections/Model/Collection.swift:124]: consider CollectionAuthor -> Author

**Review by tomerd — APPROVED** [review]

**Comment by neonichu** [comment]
What's the driver for using `description`? I always find it a bit overloaded, because of `CustomStringConvertible` and think using `summary` was preferable.

**Comment by yim-lee** [comment]
> because of CustomStringConvertible

Yeah that was the reason I used `package.summary` and `collection.overview` instead of `description` at the beginning, but then these _are_ descriptions, and calling them differently feels a little superficial. Also since we name them `description` in parts of the collections code I figured we should be consistent.

**Comment by neonichu** [comment]
> Also since we name them description in parts of the collections code I figured we should be consistent.

Personally, I'd prefer using something like `summary` throughout.

File: Sources/PackageCollections/Model/Collection.swift (hunk 4 above)
Comment (yim-lee) [reply Sources/PackageCollections/Model/Collection.swift:124]: Don't want to confuse that with package author.

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Review by neonichu — APPROVED** [review]

**Pull request merged**